libs/entitlement/agent/src/main/java/org/elasticsearch/entitlement/agent/EntitlementAgent.java

```diff
@@ -25,10 +25,16 @@
  */
 public class EntitlementAgent {
 
+    /**
+     * The agent main method
+     * @param agentArgs arguments passed to the agent.For our agent, this is the class to load and use for Entitlement Initialization.
+     *                  See e.g. {@code EntitlementsBootstrap#loadAgent}
+     * @param inst      The {@link Instrumentation} instance to use for injecting Entitlements checks
+     */
     public static void agentmain(String agentArgs, Instrumentation inst) {
         final Class<?> initClazz;
         try {
-            initClazz = Class.forName("org.elasticsearch.entitlement.initialization.EntitlementInitialization");
+            initClazz = Class.forName(agentArgs);
```

Comment:
Making the arguments just be the name of the class, rather than some kind of option flags, doesn't allow much evolution in the future, but then, we are in control of all of this code and change it in the future if need be, just like we did this time, so it doesn't actually matter much.

Reply:
I had exactly the same conversation in my head :D wondering if I should do that, and thinking we can change this any time anyway, so I decided to let it like this for now. Unless you or someone else feels strongly about this.